pkg/distributed/hosts/resolver.go

```diff
@@ -0,0 +1,15 @@
+package hosts
+
+import "context"
+
+// ID identifies a remote goProbe instance to query (host, FQDN, UUID, etc.).
+// If you already have an ID type elsewhere, feel free to replace this.
+type ID string
```

Comment:
Unless you define receiver methods on an `ID` this could be a type alias for `string`, that way you can save yourself the clumsy cast to `string(host)` further down...